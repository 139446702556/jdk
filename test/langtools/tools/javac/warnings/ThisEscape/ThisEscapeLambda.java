/*
 * @test /nodynamiccopyright/
 * @bug 8015831
 * @compile/ref=ThisEscapeLambda.out -Xlint:this-escape -XDrawDiagnostics ThisEscapeLambda.java
 * @summary Verify 'this' escape detection properly handles lambdas
 */

public class ThisEscapeLambda {

    public ThisEscapeLambda() {
        Runnable r = () -> {
            this.mightLeak();
        };
        System.out.println(r);
    }

    public void mightLeak() {
    }
}
