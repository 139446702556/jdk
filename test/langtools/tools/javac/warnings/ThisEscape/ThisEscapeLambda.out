ThisEscapeLambda.java:14:28: compiler.warn.possible.this.escape
1 warning
